Enable dashboard filtering tests and run 20x

stacey-gammon opened the pull request.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/980/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1079/)

Stacey-Gammon commented:
Triggered the flaky error.  Now to search for a fix....

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1105/)

Stacey-Gammon commented:
Hmm, something isn't right in the original logic, since it still fails too quickly:

```
16:07:22                │ debg Renderable.waitForRender for 17 elements
16:07:22                │ debg in findAllByCssSelector: [data-render-complete="true"]
16:07:23                │ debg --- retry.try error: 16 elements completed rendering, waiting on a total of 17
16:07:23                │ debg in findAllByCssSelector: [data-render-complete="true"]
16:07:23                │ debg --- retry.try failed again with the same message...
16:07:24                │ debg in findAllByCssSelector: [data-render-complete="true"]
16:07:24                │ debg --- retry.try failed again with the same message...
```

Each retry should be taking longer to fail... 😕

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1171/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1345/)

Stacey-Gammon commented:
jenkins, test this 

some random error.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1357/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1383/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1390/)

Stacey-Gammon commented:
Looks like this is now failing due to https://github.com/elastic/kibana/issues/22561

Stacey-Gammon commented:
retest

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1605/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1623/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1668/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1709/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1734/)

Stacey-Gammon commented:
Latest logs is interesting:

```
    │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: timelion split 5 on bytes: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: tag cloud with not 0 bytes filter: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: tsvb top n with bytes filter: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: Goal unique count: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: tsvb time series with bytes filter split by clientip: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: min bytes metric with query: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: max bytes in US - area chart with filter: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: tsvb markdown: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: max bytes guage percent mode: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 34688:20 "Emitting timelionChartRendered"
               │      http://localhost:5620/bundles/commons.bundle.js 81823:12 "Calling renderComplete"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: input control with filter: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: standard deviation heatmap with other bucket: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: tsvb metric with custom interval and bytes filter: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: split by geo with query: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: vega: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: Data table top hit with significant terms geo.src: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: tsvb top n with bytes filter: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: tsvb time series with bytes filter split by clientip: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: min bytes metric with query: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: Goal unique count: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: max bytes in US - area chart with filter: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: tsvb markdown: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: max bytes guage percent mode: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: geo map with filter and query bytes > 5000 in US geo.src, heatmap setting: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: tag cloud with not 0 bytes filter: data-render-complete: true"
```

You see `"Filter Bytes Test: timelion split 5 on bytes: data-render-complete: false"`

And you see ` "Emitting timelionChartRendered"`

But the render complete helper never registers that and sets to true!

cc @elastic/kibana-visualizations

Stacey-Gammon commented:
From my local run, the event goes through immediately:

```
  http://localhost:5620/bundles/commons.bundle.js 140535:20 "Filter Bytes Test: timelion split 5 on bytes: data-render-complete: false"
           │      http://localhost:5620/bundles/commons.bundle.js 140525:20 "undefined: Render complete setup"
           │      http://localhost:5620/bundles/commons.bundle.js 140535:20 "Filter Bytes Test: max bytes guage percent mode: data-render-complete: false"
           │      http://localhost:5620/bundles/commons.bundle.js 140540:20 "Filter Bytes Test: tsvb markdown: data-render-complete: true"
           │      http://localhost:5620/bundles/commons.bundle.js 140535:20 "Filter Bytes Test: Goal unique count: data-render-complete: false"
           │      http://localhost:5620/bundles/commons.bundle.js 140540:20 "Filter Bytes Test: input control with filter: data-render-complete: true"
           │      http://localhost:5620/bundles/commons.bundle.js 140535:20 "Filter Bytes Test: max bytes guage percent mode: data-render-complete: false"
           │      http://localhost:5620/bundles/commons.bundle.js 140540:20 "Filter Bytes Test: Goal unique count: data-render-complete: true"
           │      http://localhost:5620/bundles/commons.bundle.js 140525:20 "undefined: Render complete setup"
           │      http://localhost:5620/bundles/commons.bundle.js 140535:20 "Filter Bytes Test: min bytes metric with query: data-render-complete: false"
           │      http://localhost:5620/bundles/kibana.bundle.js 115917:24 "Setting timelion rendercomplete to false, props: "
           │      http://localhost:5620/bundles/commons.bundle.js 140535:20 "Filter Bytes Test: timelion split 5 on bytes: data-render-complete: false"
           │      http://localhost:5620/bundles/commons.bundle.js 140540:20 "Filter Bytes Test: max bytes guage percent mode: data-render-complete: true"
           │      http://localhost:5620/bundles/commons.bundle.js 140525:20 "undefined: Render complete setup"
           │      http://localhost:5620/bundles/commons.bundle.js 140535:20 "Filter Bytes Test: split by geo with query: data-render-complete: false"
           │      http://localhost:5620/bundles/commons.bundle.js 357155:10 "TimelionVisController"
           │      http://localhost:5620/bundles/commons.bundle.js 274343:16 "Rendering timelion chart"
           │      http://localhost:5620/bundles/commons.bundle.js 274646:20 "Emitting timelionChartRendered"
           │      http://localhost:5620/bundles/commons.bundle.js 357160:12 "Calling renderComplete"
           │      http://localhost:5620/bundles/commons.bundle.js 140540:20 "Filter Bytes Test: timelion split 5 on bytes: data-render-complete: true"

```

Stacey-Gammon commented:
I have a theory.  The renderSubscription in timelion is created in the react component's constructor but removed in the `componentWillUnmount`.  In react docs:

```
Avoid introducing any side-effects or subscriptions in the constructor. For those use cases, use componentDidMount() instead.
```

I think perhaps the renderSubscription should be set up in componentDidMount...  

Will test my theory

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1818/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1820/)

Stacey-Gammon commented:
dang. theory not right. still no pass
```
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: timelion split 5 on bytes: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: input control with filter: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: tsvb markdown: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: standard deviation heatmap with other bucket: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: Goal unique count: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: vega: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: tsvb metric with custom interval and bytes filter: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: max bytes in US - area chart with filter: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: tsvb time series with bytes filter split by clientip: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: tag cloud with not 0 bytes filter: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: tsvb top n with bytes filter: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: split by geo with query: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: Data table top hit with significant terms geo.src: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: max bytes guage percent mode: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: min bytes metric with query: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: input control with filter: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: tsvb markdown: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: standard deviation heatmap with other bucket: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: vega: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: tsvb metric with custom interval and bytes filter: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: Goal unique count: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: max bytes in US - area chart with filter: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: tsvb time series with bytes filter split by clientip: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: tsvb top n with bytes filter: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: split by geo with query: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: Data table top hit with significant terms geo.src: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: geo map with filter and query bytes > 5000 in US geo.src, heatmap setting: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 34688:20 "Emitting timelionChartRendered"
               │      http://localhost:5620/bundles/commons.bundle.js 81823:12 "Calling renderComplete"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: timelion split 5 on bytes: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: tag cloud with not 0 bytes filter: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22624:20 "undefined: Render complete setup"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: geo map with filter and query bytes > 5000 in US geo.src, heatmap setting: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: max bytes guage percent mode: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: min bytes metric with query: data-render-complete: false"
               │      http://localhost:5620/bundles/kibana.bundle.js 111046:24 "Timelion: setting up renderSubject: "
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: timelion split 5 on bytes: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: input control with filter: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: tsvb markdown: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: standard deviation heatmap with other bucket: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: Goal unique count: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: vega: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: tsvb metric with custom interval and bytes filter: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: max bytes in US - area chart with filter: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: tsvb time series with bytes filter split by clientip: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: tag cloud with not 0 bytes filter: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: tsvb top n with bytes filter: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: split by geo with query: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: Data table top hit with significant terms geo.src: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 34688:20 "Emitting timelionChartRendered"
               │      http://localhost:5620/bundles/commons.bundle.js 81823:12 "Calling renderComplete"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: max bytes guage percent mode: data-render-complete: true"
```

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1836/)

Stacey-Gammon commented:
Logs show a `emitting timelionSheetRendered` before the render call in `Timelion: in siwtch map, vis is `, but the render never resolves.

```
 http://localhost:5620/bundles/kibana.bundle.js 111055:24 "Timelion: this.renderSubject.next"
             │      http://localhost:5620/bundles/kibana.bundle.js 110959:28 "Timelion: dispatchRenderStart"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: timelion: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: goal: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: markdown: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: vega: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: input control: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: tsvb-table: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: non timebased line chart - dog data - with filter: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: tag cloud: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: bar: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 34688:20 "Emitting timelionChartRendered"
             │      http://localhost:5620/bundles/commons.bundle.js 81825:12 "Calling renderComplete"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: scripted filter and query: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: tsvb-guage: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 72232:16 "AngularVisController.render"
             │      http://localhost:5620/bundles/commons.bundle.js 72235:20 "AngularVisController.updateScope"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: datatable: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: animal sounds pie: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: input control parent: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: metric: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: tsvb-markdown: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: heatmap: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: tsvb-topn: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: tsvb-metric: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: area with not filter: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: animal weights linked to search: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: tsvb-ts: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Filter Bytes Test: vega: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: guage: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: pie: data-render-complete: true"
             │      http://localhost:5620/bundles/kibana.bundle.js 110976:32 "Timelion: in siwtch map, vis is "
             │      http://localhost:5620/bundles/commons.bundle.js 72232:16 "AngularVisController.render"
             │      http://localhost:5620/bundles/commons.bundle.js 72235:20 "AngularVisController.updateScope"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: markdown: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: vega: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: input control: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: tsvb-table: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: goal: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: non timebased line chart - dog data - with filter: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: bar: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: region map: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: geo map: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "Rendering Test: tag cloud: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22624:20 "undefined: Render complete setup"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:20 "{{sharedItemTitle}}: data-render-complete: true"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: scripted filter and query: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: tsvb-guage: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: datatable: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: animal sounds pie: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: region map: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: input control parent: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: metric: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: tsvb-markdown: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: heatmap: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: tsvb-topn: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: tsvb-metric: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: geo map: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: area with not filter: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: animal weights linked to search: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: tsvb-ts: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Filter Bytes Test: vega: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: guage: data-render-complete: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: pie: data-render-complete: false"
             │      http://localhost:5620/bundles/kibana.bundle.js 111055:24 "Timelion: this.renderSubject.next"
             │      http://localhost:5620/bundles/kibana.bundle.js 110959:28 "Timelion: dispatchRenderStart"
             │      http://localhost:5620/bundles/commons.bundle.js 22634:20 "Rendering Test: timelion: data-render-complete: false"
```

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1867/)

Stacey-Gammon commented:
F'ing gradle failure:

```
20:08:21    │ERROR 
20:08:21    │ERROR FAILURE: Build failed with an exception.
20:08:21    │ERROR 
20:08:21    │ERROR * Where:
20:08:21    │ERROR Build file '/var/lib/jenkins/workspace/elastic+kibana+pull-request+multijob-selenium/elasticsearch/build.gradle' line: 30
20:08:21    │ERROR 
20:08:21    │ERROR * What went wrong:
20:08:21    │ERROR Error resolving plugin [id: 'com.gradle.build-scan', version: '1.13.2']
```

jenkins, test this

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1917/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1928/)

Stacey-Gammon commented:
Looks like it passes now, with the added logs:

```
  http://localhost:5620/bundles/kibana.bundle.js 111063:24 "VisualizationChart: this.renderSubject.next"
             │      http://localhost:5620/bundles/kibana.bundle.js 110959:28 "VisualizationChart: dispatchRenderStart"
             │      http://localhost:5620/bundles/commons.bundle.js 22639:24 "RenderCompleteHelper: Rendering Test: timelion: data-render-complete: false"
             │      http://localhost:5620/bundles/kibana.bundle.js 110986:40 "VisualizationChart: Calling this.visualization.render"
             │      http://localhost:5620/bundles/commons.bundle.js 72306:16 "AngularVisController: render (Rendering Test: timelion)"
             │      http://localhost:5620/bundles/commons.bundle.js 72312:20 "AngularVisController.updateScope: (Rendering Test: timelion) resetting scope.renderComplete"
             │      http://localhost:5620/bundles/commons.bundle.js 72306:16 "AngularVisController: render (Rendering Test: datatable)"
             │      http://localhost:5620/bundles/commons.bundle.js 72312:20 "AngularVisController.updateScope: (Rendering Test: datatable) resetting scope.renderComplete"
             │      http://localhost:5620/bundles/commons.bundle.js 34696:20 "Timelion Schema: Emitting timelionChartRendered"
             │      http://localhost:5620/bundles/commons.bundle.js 81899:12 "TimelionVisController: Calling scope.renderComplete"
             │      http://localhost:5620/bundles/kibana.bundle.js 110992:40 "VisualizationChart: Returned from this.visualization.render"
             │      http://localhost:5620/bundles/kibana.bundle.js 111005:24 "VisualizationChart: success$ merged, chartDiv.current: false"
             │      http://localhost:5620/bundles/commons.bundle.js 22646:24 "RenderCompleteHelper: Rendering Test: timelion: data-render-complete: true"
```

Just failing due to https://github.com/elastic/kibana/issues/22783

Stacey-Gammon commented:
Removed the two logs inside the timelion `chart.js` and hit the error again:

```
               │ debg DashboardAddPanel.closeAddPanel
               │ debg DashboardAddPanel.isAddPanelOpen
               │ debg TestSubjects.exists(dashboardAddPanel)
               │ debg existsByDisplayedByCssSelector [data-test-subj~="dashboardAddPanel"]
               │ debg isGlobalLoadingIndicatorVisible
               │ debg TestSubjects.exists(globalLoadingIndicator)
               │ debg existsByDisplayedByCssSelector [data-test-subj~="globalLoadingIndicator"]
               │ debg awaitGlobalLoadingIndicatorHidden
               │ debg TestSubjects.find(globalLoadingIndicator-hidden)
               │ debg findByCssSelector [data-test-subj~="globalLoadingIndicator-hidden"]
               │ debg about to wait for first render complete
               │ info Browser logs:
               │       http://localhost:5620/bundles/commons.bundle.js 22623:20 "Rendering Test: animal sounds pie: Render complete destroy"
               │      http://localhost:5620/bundles/commons.bundle.js 22623:20 "Filter Test: animals: linked to search with filter: Render complete destroy"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:24 "RenderCompleteHelper: Filter Bytes Test: timelion split 5 on bytes: data-render-complete: false"
               │      http://localhost:5620/bundles/kibana.bundle.js 111042:20 "VisualizationChart: componentDidUpdate"
               │      http://localhost:5620/bundles/kibana.bundle.js 111063:24 "VisualizationChart: this.renderSubject.next"
               │      http://localhost:5620/bundles/kibana.bundle.js 110959:28 "VisualizationChart: dispatchRenderStart"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:24 "RenderCompleteHelper: Filter Bytes Test: timelion split 5 on bytes: data-render-complete: false"
               │      http://localhost:5620/bundles/kibana.bundle.js 110986:40 "VisualizationChart: Calling this.visualization.render"
               │      http://localhost:5620/bundles/commons.bundle.js 72306:16 "AngularVisController: render (Filter Bytes Test: timelion split 5 on bytes)"
               │      http://localhost:5620/bundles/commons.bundle.js 72309:20 "AngularVisController.updateScope: (Filter Bytes Test: timelion split 5 on bytes)"
               │      http://localhost:5620/bundles/commons.bundle.js 81894:10 "TimelionVisController"
               │      http://localhost:5620/bundles/commons.bundle.js 34393:16 "Rendering timelion chart"
               │      http://localhost:5620/bundles/commons.bundle.js 34696:20 "Timelion Schema: Emitting timelionChartRendered"
               │      http://localhost:5620/bundles/commons.bundle.js 81899:12 "TimelionVisController: Calling scope.renderComplete"
               │      http://localhost:5620/bundles/kibana.bundle.js 110992:40 "VisualizationChart: Returned from this.visualization.render"
               │      http://localhost:5620/bundles/kibana.bundle.js 111005:24 "VisualizationChart: success$ merged, chartDiv.current: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22646:24 "RenderCompleteHelper: Filter Bytes Test: timelion split 5 on bytes: data-render-complete: true"
               │      http://localhost:5620/bundles/kibana.bundle.js 111042:20 "VisualizationChart: componentDidUpdate"
               │      http://localhost:5620/bundles/kibana.bundle.js 111063:24 "VisualizationChart: this.renderSubject.next"
               │      http://localhost:5620/bundles/kibana.bundle.js 110959:28 "VisualizationChart: dispatchRenderStart"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:24 "RenderCompleteHelper: Filter Bytes Test: timelion split 5 on bytes: data-render-complete: false"
               │      http://localhost:5620/bundles/kibana.bundle.js 110986:40 "VisualizationChart: Calling this.visualization.render"
               │      http://localhost:5620/bundles/commons.bundle.js 72306:16 "AngularVisController: render (Filter Bytes Test: timelion split 5 on bytes)"
               │      http://localhost:5620/bundles/commons.bundle.js 72309:20 "AngularVisController.updateScope: (Filter Bytes Test: timelion split 5 on bytes)"
               │      http://localhost:5620/bundles/commons.bundle.js 34524:18 "Destroying timelion chart"
               │      http://localhost:5620/bundles/commons.bundle.js 34393:16 "Rendering timelion chart"
               │      http://localhost:5620/bundles/commons.bundle.js 34696:20 "Timelion Schema: Emitting timelionChartRendered"
               │      http://localhost:5620/bundles/commons.bundle.js 81899:12 "TimelionVisController: Calling scope.renderComplete"
               │      http://localhost:5620/bundles/kibana.bundle.js 110992:40 "VisualizationChart: Returned from this.visualization.render"
               │      http://localhost:5620/bundles/kibana.bundle.js 111005:24 "VisualizationChart: success$ merged, chartDiv.current: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22646:24 "RenderCompleteHelper: Filter Bytes Test: timelion split 5 on bytes: data-render-complete: true"
               │      http://localhost:5620/bundles/kibana.bundle.js 111063:24 "VisualizationChart: this.renderSubject.next"
               │      http://localhost:5620/bundles/kibana.bundle.js 110959:28 "VisualizationChart: dispatchRenderStart"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:24 "RenderCompleteHelper: Filter Bytes Test: timelion split 5 on bytes: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 72309:20 "AngularVisController.updateScope: (Filter Bytes Test: Data table top hit with significant terms geo.src)"
               │      http://localhost:5620/bundles/kibana.bundle.js 110986:40 "VisualizationChart: Calling this.visualization.render"
               │      http://localhost:5620/bundles/commons.bundle.js 72306:16 "AngularVisController: render (Filter Bytes Test: timelion split 5 on bytes)"
               │      http://localhost:5620/bundles/commons.bundle.js 72309:20 "AngularVisController.updateScope: (Filter Bytes Test: timelion split 5 on bytes)"
               │      http://localhost:5620/bundles/commons.bundle.js 34696:20 "Timelion Schema: Emitting timelionChartRendered"
               │      http://localhost:5620/bundles/commons.bundle.js 81899:12 "TimelionVisController: Calling scope.renderComplete"
               │      http://localhost:5620/bundles/kibana.bundle.js 110992:40 "VisualizationChart: Returned from this.visualization.render"
               │      http://localhost:5620/bundles/kibana.bundle.js 111005:24 "VisualizationChart: success$ merged, chartDiv.current: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22646:24 "RenderCompleteHelper: Filter Bytes Test: timelion split 5 on bytes: data-render-complete: true"
               │      http://localhost:5620/bundles/kibana.bundle.js 111063:24 "VisualizationChart: this.renderSubject.next"
               │      http://localhost:5620/bundles/kibana.bundle.js 110959:28 "VisualizationChart: dispatchRenderStart"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:24 "RenderCompleteHelper: Filter Bytes Test: timelion split 5 on bytes: data-render-complete: false"
               │      http://localhost:5620/bundles/kibana.bundle.js 110986:40 "VisualizationChart: Calling this.visualization.render"
               │      http://localhost:5620/bundles/commons.bundle.js 72306:16 "AngularVisController: render (Filter Bytes Test: timelion split 5 on bytes)"
               │      http://localhost:5620/bundles/commons.bundle.js 72309:20 "AngularVisController.updateScope: (Filter Bytes Test: timelion split 5 on bytes)"
               │      http://localhost:5620/bundles/commons.bundle.js 34696:20 "Timelion Schema: Emitting timelionChartRendered"
               │      http://localhost:5620/bundles/commons.bundle.js 81899:12 "TimelionVisController: Calling scope.renderComplete"
               │      http://localhost:5620/bundles/kibana.bundle.js 110992:40 "VisualizationChart: Returned from this.visualization.render"
               │      http://localhost:5620/bundles/kibana.bundle.js 111005:24 "VisualizationChart: success$ merged, chartDiv.current: false"
               │      http://localhost:5620/bundles/commons.bundle.js 22646:24 "RenderCompleteHelper: Filter Bytes Test: timelion split 5 on bytes: data-render-complete: true"
               │      http://localhost:5620/bundles/kibana.bundle.js 111063:24 "VisualizationChart: this.renderSubject.next"
               │      http://localhost:5620/bundles/kibana.bundle.js 110959:28 "VisualizationChart: dispatchRenderStart"
               │      http://localhost:5620/bundles/commons.bundle.js 22639:24 "RenderCompleteHelper: Filter Bytes Test: timelion split 5 on bytes: data-render-complete: false"
               │      http://localhost:5620/bundles/commons.bundle.js 34696:20 "Timelion Schema: Emitting timelionChartRendered"
               │      http://localhost:5620/bundles/commons.bundle.js 81899:12 "TimelionVisController: Calling scope.renderComplete"
               │      http://localhost:5620/bundles/kibana.bundle.js 110986:40 "VisualizationChart: Calling this.visualization.render"
               │      http://localhost:5620/bundles/commons.bundle.js 72306:16 "AngularVisController: render (Filter Bytes Test: timelion split 5 on bytes)"
               │      http://localhost:5620/bundles/commons.bundle.js 72309:20 "AngularVisController.updateScope: (Filter Bytes Test: timelion split 5 on bytes)"
               │
               │ debg DashboardPage.waitForRenderComplete
               │ debg in getSharedItemsCount
               │ debg findByCssSelector [data-shared-items-count]
               │ debg Renderable.waitForRender for 17 elements
               │ debg find.allByCssSelector(selector: [data-render-complete="true"], timeout: 10000)
               │ debg --- retry.try error: 16 elements completed rendering, waiting on a total of 17
               │ debg find.allByCssSelector(selector: [data-render-complete="true"], timeout: 10000)
`

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/1944/)
